src/tests/ftest/container/auto_oc_selection.py

```diff
@@ -0,0 +1,100 @@
+#!/usr/bin/python
+"""
+  (C) Copyright 2020-2022 Intel Corporation.
+
+  SPDX-License-Identifier: BSD-2-Clause-Patent
+"""
+from apricot import TestWithServers
+from test_utils_container import TestContainer
+from avocado.core.exceptions import TestFail
+
+
+class AutoOCSelectionTest(TestWithServers):
+    """Automatic object class selection test.
+
+    Test Class Description:
+    Create a container with different RF and OC combinations. We expect some
+    combinations to fail due to the restriction set by DAOS. (This is more like
+    auto OC "restriction" rather than auto OC "selection".)
+
+    :avocado: recursive
+    """
+    def test_oc_restrictions(self):
+        """JIRA ID: DAOS-7595
+
+        Test Description:
+        Create a container with RF 0 to 4 - OC S1, RP_2G1, RP_3G1, RP_4G1, and
+        RP_5G1.
+
+        Use Cases:
+        Create a pool. Then create each container and verify against its
+        failure expected state.
+
+        :avocado: tags=all,full_regression
+        :avocado: tags=vm
+        :avocado: tags=container,oc_restrictions
```

Comment:
Suggestion/Minor/Optional - the "features" tag is usually on a separate line
```suggestion
        :avocado: tags=all,full_regression
        :avocado: tags=vm
        :avocado: tags=container
        :avocado: tags=oc_restrictions
```

Reply:
Done